### PR TITLE
(DOCSP-29612): Update Relationship model examples for v12

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-29612

### Staged Changes

- [Relationships & Embedded Objects](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-29612/sdk/node/model-data/relationships-and-embedded-objects/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/node/v12/__tests__/relationships.test.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
+import Realm from "realm";
+
+describe("Define Relationship Properties", () => {
+  test("should define a to-one relationship", async () => {
+    // :snippet-start: define-one-to-one
+    class Manufacturer extends Realm.Object {
+      _id!: Realm.BSON.ObjectId;
+      name!: String;
```
Comment (elle-j) [inline examples/node/v12/__tests__/relationships.test.ts:8]:
```suggestion
      name!: string;
```

File: examples/node/v12/__tests__/relationships.test.ts [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+    // close the realm
+    realm.close();
+  });
+
+  test("should define a to-many relationship", async () => {
+    // :snippet-start: define-one-to-many
+    class Manufacturer extends Realm.Object {
+      _id!: Realm.BSON.ObjectId;
+      name!: String;
```
Comment (elle-j) [inline examples/node/v12/__tests__/relationships.test.ts:86]:
```suggestion
      name!: string;
```

File: examples/node/v12/__tests__/relationships.test.ts [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+
+    // close the realm
+    realm.close();
+  });
+
+  test("should define an inverse relationship using model classes", async () => {
+    // :snippet-start: define-inverse
+    class Manufacturer extends Realm.Object {
+      _id!: Realm.BSON.ObjectId;
+      name!: String;
```
Comment (elle-j) [inline examples/node/v12/__tests__/relationships.test.ts:178]:
```suggestion
      name!: string;
```

File: examples/node/v12/__tests__/relationships.test.ts [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+
+    // close the realm
+    realm.close();
+  });
+
+  test("should define an embedded object property using model classes", async () => {
+    // :snippet-start: define-embedded-property-model-classes
+    class Manufacturer extends Realm.Object {
+      _id!: Realm.BSON.ObjectId;
+      name!: String;
```
Comment (elle-j) [inline examples/node/v12/__tests__/relationships.test.ts:267]:
```suggestion
      name!: string;
```

File: source/examples/generated/node/v12/relationships.test.snippet.define-embedded-property-model-classes.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
+class Manufacturer extends Realm.Object {
+  _id!: Realm.BSON.ObjectId;
+  name!: String;
```
Comment (elle-j) [inline source/examples/generated/node/v12/relationships.test.snippet.define-embedded-property-model-classes.ts:3]:
```suggestion
  name!: string;
```

File: source/examples/generated/node/v12/relationships.test.snippet.define-inverse.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+class Manufacturer extends Realm.Object {
+  _id!: Realm.BSON.ObjectId;
+  name!: String;
```
Comment (elle-j) [inline source/examples/generated/node/v12/relationships.test.snippet.define-inverse.ts:3]:
```suggestion
  name!: string;
```

File: source/examples/generated/node/v12/relationships.test.snippet.define-one-to-many.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
+class Manufacturer extends Realm.Object {
+  _id!: Realm.BSON.ObjectId;
+  name!: String;
```
Comment (elle-j) [inline source/examples/generated/node/v12/relationships.test.snippet.define-one-to-many.ts:3]:
```suggestion
  name!: string;
```

File: source/examples/generated/node/v12/relationships.test.snippet.define-one-to-one.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+class Manufacturer extends Realm.Object {
+  _id!: Realm.BSON.ObjectId;
+  name!: String;
```
Comment (elle-j) [inline source/examples/generated/node/v12/relationships.test.snippet.define-one-to-one.ts:3]:
```suggestion
  name!: string;
```

File: source/examples/generated/node/v12/relationships.test.snippet.define-inverse.ts (hunk 6 above)
Comment (dacharyc) [reply source/examples/generated/node/v12/relationships.test.snippet.define-inverse.ts]: Thank you so much for catching these! I work more often in Swift where these are uppercased, and I'm used to the IDE complaining if I get the casing wrong.

For future PR reviews, no need to suggest changes to any files in the `source/examples/generated` directory. We use a tool to automatically generate these examples from the markup in the `examples/node/v12/__tests__/` directory. So if you suggest corrections there, they should be automatically applied to the generated examples when we re-generate them.

Thanks again for your help!

Comment (elle-j) [reply source/examples/generated/node/v12/relationships.test.snippet.define-inverse.ts]: Ah okay good to know! 🙂

File: examples/node/v12/__tests__/relationships.test.ts [hunk 9]
```diff
@@ -0,0 +1,324 @@
+import Realm from "realm";
```
Comment (krollins-mdb) [inline examples/node/v12/__tests__/relationships.test.ts:1]: Similar to #2980, let's use top-level named imports as that's a new feature in v12 and how most TypeScript/JavaScript devs prefer to work.
```suggestion
import Realm, {BSON} from "realm";
```

Comment (krollins-mdb) [reply examples/node/v12/__tests__/relationships.test.ts:1]: Then, we can replace all instances of `Realm.BSON` with just `BSON`.

File: examples/node/v12/__tests__/relationships.test.ts [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+      });
+    });
+
+    const manufacturers: Realm.Results<Manufacturer> =
+      realm.objects(Manufacturer);
+
+    const manufacturerCars = manufacturers[0].cars;
+
+    // :snippet-start: obtain-inverse-relationship-dynamically
+    // Get the Manufacturer who makes the Car
```
Comment (krollins-mdb) [inline examples/node/v12/__tests__/relationships.test.ts:132]: Is this comment supposed to go before line 134, not 133? Line 133 is getting all Car object models - not the linked manufacturer.

File: source/sdk/node/model-data/relationships-and-embedded-objects.txt [hunk 11]
```diff
@@ -114,31 +131,38 @@ You can dynamically retrieve an object with an inverse relationship without
...
 .. example::
    
-   In the following continuation from the inverse relationship example, we have removed the
-   ``assignee`` field with type 'linkingObjects' from the Task schema. An application developer
-   creates several users and task objects, and the application pushes the newly created tasks into a 
-   user's task field. 
+   In the following continuation from the inverse relationship example, we 
+   have removed the ``manufacturer`` field with type 'linkingObjects' from 
+   the ``Car`` schema. An application developer creates several manufacturers 
+   and car objects, and the application pushes the newly created cars into a
```
Comment (krollins-mdb) [inline source/sdk/node/model-data/relationships-and-embedded-objects.txt:142]:
```suggestion
   and car objects, and the application pushes the newly-created cars into a 
```

File: source/sdk/node/model-data/relationships-and-embedded-objects.txt [hunk 12]
```diff
@@ -114,31 +131,38 @@ You can dynamically retrieve an object with an inverse relationship without
...
+   have removed the ``manufacturer`` field with type 'linkingObjects' from 
+   the ``Car`` schema. An application developer creates several manufacturers 
+   and car objects, and the application pushes the newly created cars into a 
+   manufacturer's ``cars`` field. 
    
-   To find the user who owns a specific task object, the developer calls the
-   ``task.linkingObjects()`` method and passes the "User" class name and "tasks" field as parameters. 
+   To find the manufacturer who makes a specific car object, the developer 
+   calls the ``.linkingObjects()`` method and passes the "Manufacturer" 
+   class name and "cars" field as parameters.
```
Comment (krollins-mdb) [inline source/sdk/node/model-data/relationships-and-embedded-objects.txt:147]: I know this is old wording here (not yours), but this style drives me crazy.

```suggestion
   To find the manufacturer who makes a specific car object, call ``.linkingObjects()``
   and pass the "Manufacturer" class name and "cars" field as parameters.
```

**Review by krollins-mdb — APPROVED** [review]
Thank you for making this page better! I have a few minor suggestions. Other than those, LGTM!

File: examples/node/v12/__tests__/relationships.test.ts (hunk 9 above)
Comment (dacharyc) [reply examples/node/v12/__tests__/relationships.test.ts:1]: Thanks for catching this! I meant to update it after I updated the other PR, but got pulled away and had forgotten by the time I got back to it.

**Pull request merged**